src/lib/registration/autoyast_addons.rb

```diff
@@ -64,9 +66,18 @@ def register
     def select_addons(all_addons)
       # select the requested addons from the AY profile
       requested_addons.each do |addon|
-        server_addon = all_addons.find do |a|
-          a.identifier == addon["name"] && a.version == addon["version"] &&
-            a.arch == addon["arch"]
+        # Set architecture if it is not defined in the requested addon
+        requested_arch = addon["arch"] || Yast::Arch.architecture
```

Comment:
Unfortunately you cannot fallback to `Yast::Arch.architecture`, that does not work correctly in some cases. E.g. Arch can return `s390_64` while SCC expects `s390x`.

Use the [Y2Packager::ProductControlProduct::REG_ARCH](https://github.com/yast/yast-yast2/blob/master/library/packages/src/lib/y2packager/product_control_product.rb#L35) mapping like [this](https://github.com/yast/yast-yast2/blob/master/library/packages/src/lib/y2packager/product_control_product.rb#L61).